File: 0007-reverse-integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        a = 0
        negative = False
        if x<0:
            x = abs(x)
            negative = True
        while(x>0):
            a = (a*10) + (x%10)
            x = x//10
        if (-2)**31<a<(2**31)-1:
            if negative:
                return -a
            else:
                return a
        else:
            return 0

File: 0007-reverse-integer/solution.cs
public class Solution {
    public int Reverse(int x) {
        int reversed = 0;
        while (x != 0) {
            int pop = x % 10;
            x /= 10;
            if (reversed > Int32.MaxValue / 10 || (reversed == Int32.MaxValue / 10 && pop > 7)) return 0;
            if (reversed < Int32.MinValue / 10 || (reversed == Int32.MinValue / 10 && pop < -8)) return 0;
            reversed = reversed * 10 + pop;
        }

        return reversed;
    }
}

